gitlab.go

```diff
@@ -856,6 +868,12 @@ func (r *Response) populatePageValues() {
 // interface, the raw response body will be written to v, without attempting to
 // first decode it.
 func (c *Client) Do(req *retryablehttp.Request, v interface{}) (*Response, error) {
+	// If not yet configured, try to configure the rate limiter. Fail
+	// silently as the limiter will be disabled in case of an error.
+	if c.limiter == nil {
+		c.configureLimiter()
```

Comment:
This is a data race: the `Do()` method might be called concurrently from multiple goroutines and the `c.limiter` read and write operations are not synchronized properly. `sync.Once` might be a good way to perform the synchronization here.

Reply:
Yep, good on... I didn't know if `sync.Once` also blocked any other calls until the one execution has finished, but looking at the implementation is indeed does that (by using a lock), so I'll use a `sync.Once` instead 👍